main.cpp

```diff
@@ -78,7 +79,8 @@ int main(int argc, char *args[])
             printf("Window could not be created! SDL_Error: %s\n",
                    SDL_GetError());
         } else {
-            Game game;
+            Map default_map(g_map);
+            Game game(&default_map);
```

Comment:
You can create an instance of Game object instead.

Reply:
Sorry, I'm not sure what you mean. Do you mean to create the instance of Map inside Game instead of creating it then pass it to Game?

Reply:
To encapsulate the transition and internal state of Game object, the Map should be initialized and tracked within the lifecycle of Game object.